### PR TITLE
add test for disable checksumoffload on IPIP device

### PR DESCRIPTION
**Pull request by cyclinder**
## Description

add test for disable checksumoffload on IPIP device

## Related issues/PRs

https://github.com/projectcalico/calico/pull/8031#pullrequestreview-1746597070

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/fv/ipip_test.go [hunk 1]
```diff
@@ -129,15 +128,6 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ IPIP topology before adding
 		infra.Stop()
 	})
 
-	It("should use the --random-fully flag in the MASQUERADE rules", func() {
```
Comment (caseydavenport) [inline felix/fv/ipip_test.go:132]: Is this intentional? Do we need to remove this test as part of the change?

Comment (cyclinder) [reply felix/fv/ipip_test.go]: I'm sorry, it's a typo here, Now this has been fixed.

File: felix/fv/ipip_test.go [hunk 2]
```diff
@@ -65,8 +65,7 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ IPIP topology before adding
 		if BPFMode() && getDataStoreType(infra) == "etcdv3" {
 			Skip("Skipping BPF test for etcdv3 backend.")
 		}
-		tc, client = infrastructure.StartNNodeTopology(2, infrastructure.DefaultTopologyOptions(), infra)
-
+		tc, client = infrastructure.StartNNodeTopology(2, createDefaultTopologyOptionsWithChecksumOffload(), infra)
```
[the remark this answers is not in the document]
Comment (tomastigera) [reply felix/fv/ipip_test.go:68]: If the `infrastructure.StartNNodeTopology` took functional options, I'd go with this. However, isn't it easier and more like what the other tests do, that they get default options and then set what is necessary like say here https://github.com/projectcalico/calico/blob/master/felix/fv/bpf_test.go#L339-L346 In fact, that is what your `createDefaultTopologyOptionsWithChecksumOffload` does inside.

**Review by tomastigera — COMMENTED** [review]
What it the motivation for enabling csum offload in the tests? Note that csums are mostly ignored in these local/docker environments.

Do you have a case where these checksums are broken? We are currently looking into the broken csum issue and if you had such an environment, I would have more than happy to see your repro.

**Comment by caseydavenport** [comment]
@tomastigera motivation is in the description - it's from this linked PR comment: https://github.com/projectcalico/calico/pull/8031#pullrequestreview-1746597070

**Comment by tomastigera** [comment]
@cyclinder I pinged you on calico user slack, I would be happy to hear about your setup as I am trying to repro this issues so that we can fix them without disabling the offload. I would be happy to talk! Thanks

File: felix/fv/ipip_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyclinder) [reply felix/fv/ipip_test.go:68]: updated, yes, This should pass an argument to this function, I'm surprised that the vscode IDE doesn't report the typo, It seems all go tools are not working. :(

**Comment by cyclinder** [comment]
Hi @tomastigera, Received your message. Let's discuss this in slack. :)

**Comment by cyclinder** [comment]
Hi @caseydavenport @tomastigera, Is it still necessary to merge this pr？

**Comment by caseydavenport** [comment]
@cyclinder I was about to ask Tomas the same thing! 

@tomastigera what say you?

**Comment by tomastigera** [comment]
@cyclinder now we enabled it by default so I don't think we need that. We might eventually want to make upgrades to turn it back on and that would like to test.

**Pull request closed without merging**